srcalc/src/main/java/gov/va/med/srcalc/vista/RpcVistaPatientDao.java

```diff
@@ -235,6 +268,38 @@ private void retrieveLabs(final int dfn, final Patient patient) throws ParseExce
         }
     }
     
+    private void retrieveHealthFactors(final int dfn, final Patient patient)
+    {
+        try
+        {
+            // Force a LinkedList to improve remove speed and allow for modification of the list
+            // since the doRpc uses Arrays.toList() which has a fixed size.
+            final List<String> rpcResults = new LinkedList<String> (fProcedureCaller.doRpc(
+                    fDuz, RemoteProcedure.GET_HEALTH_FACTORS, String.valueOf(dfn)));
+            // Now that we have all of the health factors, filter out any that are not present
+            // in the list provided by the NSO.
+            final ListIterator<String> iter = rpcResults.listIterator();
+            while(iter.hasNext())
+            {
+                final String healthFactor = iter.next();
+                if(VALID_HEALTH_FACTORS.get(healthFactor.split("\\^")[1]) == null)
+                {
+                    iter.remove();
+                }
+                else 
+                {
+                    iter.set(healthFactor.replace('^', ' '));
+                }
+            }
+            patient.setHealthFactors(rpcResults);
```

Comment:
This variable name `rpcResults` is misleading: by this line it is no longer what the RPC returned.